### PR TITLE
[ChubbsBunns] iP

### PR DESCRIPTION
**Pull request by ChubbsBunns**
I have done some work. Please admire my Pikachus. Thank you.

### CODE REVIEW

File: src/main/java/Addlist.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                            }
+                            else {
+                                description = description.concat(currWord);
+                                description = description.concat(" ");
+                            }
+                        }
+                        String dueDate = newScanner.nextLine();
+                        if (description.equals(""))
+                        {
+                            System.out.println(outPutUI.nullInputErrorMessage("Description cannot be empty :<, please write a description after the word deadline"));
```
Comment (jinxuan-owyong) [inline src/main/java/Addlist.java]: Perhaps line length should be no longer than 120 chars.

File: src/main/java/Addlist.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        Scanner newScanner = new Scanner(System.in);
+        OutputUI outPutUI = new OutputUI();
+        Marker marker = new Marker();
+        int index = 0;
+        while(true) {
+            try {
+                System.out.println("______________________________________________");
+                String input = newScanner.next();
+                System.out.println("______________________________________________");
+                switch (input) {
```
Comment (jinxuan-owyong) [inline src/main/java/Addlist.java]: Perhaps the switch statement should have the following form: Note there is no indentation for case clauses.

File: src/main/java/Addlist.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                        } catch (InputMismatchException e) {
+                            System.out.println("PIKAPII thats not a number! Please input a number after you type unmark!");
+                            newScanner.skip("[a-zA-Z1-9]*");
+                        }
+                        break;
+                    default:
+                            System.out.println(input);
+                            throw new DukeException("Pikapi is unable to find that command, please type in a correct command");
+                }
+            }catch(DukeException e) {
```
Comment (jinxuan-owyong) [inline src/main/java/Addlist.java]: Perhaps the try-catch statement should follow the recommended format (missing space)
```java
try {
    statements;
} catch (Exception exception) {
    statements;
}
```

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -1,10 +1,34 @@
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo = "_________¶¶¥¥¥¶¶¶\n"
```
Comment (jinxuan-owyong) [inline src/main/java/Duke.java]: Perhaps constant names must be all uppercase using underscore to separate words.

File: src/main/java/DukeException.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+public class DukeException extends Exception{
+
+
+
+    public DukeException()
+    {
```
Comment (jinxuan-owyong) [inline src/main/java/DukeException.java]: Perhaps you should use K&R style brackets (aka [Egyptian style](https://blog.codinghorror.com/new-programming-jargon/)).

File: src/main/java/Echo.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import java.util.Scanner;
+
+public class Echo {
+    public void Echo()
+    {
+        String surprisedPikachuFaceLogo = "⣿⣿⣿⣿⣿⡏⠉⠛⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡿⣿\n"
```
Comment (jinxuan-owyong) [inline src/main/java/Echo.java]: Perhaps constant names must be all uppercase using underscore to separate words.

**Review by jinxuan-owyong — COMMENTED** [review]
Great job on enhancing the visuals for the chatbot. However, I noticed repeated deviations from the [recommended coding standard](https://se-education.org/guides/conventions/java/basic.html), and inconsistent spacing in various lines. Consider using the keyboard shortcuts below to format your code in IntelliJ IDEA
> For Windows : Ctrl + Alt + L.
> For Ubuntu : Ctrl + Alt + Windows + L.
> For Mac : ⌥ (Option) + ⌘ (Command) + L.

File: src/main/java/Addlist.java [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        while(true) {
+            try {
+                System.out.println("______________________________________________");
+                String input = newScanner.next();
+                System.out.println("______________________________________________");
+                switch (input) {
+                    case "list":
+                        outPutUI.printList(ListOfThings, index);
+                        break;
+                    case "bye":
```
Comment (waiter-palypoo) [inline src/main/java/Addlist.java]: The case clauses and switch statement should be aligned, i.e. no indentation.

File: src/main/java/Addlist.java [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+                                outPutUI.addToListTodoMessage(newEvent, index);
+                            }
+
+                        break;
+                    case "mark":
+                            int num;
+                            try {
+                                num = newScanner.nextInt();
+                                if (index == 0) {
+                                    System.out.println("There are no items in the list! Please input an item for Pikapi :3");
```
Comment (waiter-palypoo) [inline src/main/java/Addlist.java]: Nice

File: src/main/java/Addlist.java [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+                                    break;
+                                } else if (num > index) {
+                                    System.out.println("You do not have that many items yet, Pikapi requests a number less than or equal to " + index + " after the word mark");
```
Comment (waiter-palypoo) [inline src/main/java/Addlist.java]: Good error handling

File: src/main/java/Duke.java [hunk 10]
```diff
@@ -1,10 +1,34 @@
...
+                + "________¶¶¥¥¥¥¥¶¶¶__________________________________________¶¶¶¶¶¶¶¶\n"
```
Comment (waiter-palypoo) [inline src/main/java/Duke.java]: cute pikachu

File: src/main/java/DukeException.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+        super("Pikapi is unable to find this command, please specify either todo, deadline, event, list, mark or unmark\n");
```
Comment (waiter-palypoo) [inline src/main/java/DukeException.java]: Good job considering such exceptions!

File: src/main/java/StartPikaDuke.java [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    private static final String CHAR_SPACE = " ";
+    private static final String LIST_COMMAND = "list";
+    private static final String EXIT_COMMAND = "bye";
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String MARK_COMMAND = "mark";
+    private static final String UNMARK_COMMAND = "unmark";
+    private static final String DELETE_COMMAND = "delete";
+    private static final String FIND_COMMAND = "find";
```
Comment (chinhan99) [inline src/main/java/StartPikaDuke.java]: Good use of constants here

File: src/main/java/StartPikaDuke.java [hunk 13]
```diff
@@ -0,0 +1,73 @@
+import errors.DukeException;
+import errors.ErrorMessages;
+import tasks.Task;
+import util.*;
```
Comment (chinhan99) [inline src/main/java/StartPikaDuke.java]: Refrain from using star imports.

File: src/main/java/errors/ErrorMessages.java [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+    //The following are meant for wrong command errors
+    public String provideWrongCommandText() {
+        return "Pikapi is unable to find that command, please type in a correct command";
+    }
+
+    //The following are meant for find task errors
+    public String provideNoTaskKeywordInput(){
+        return "Pikapi is unable to find a keyword to search the list for you, please add a word after the word \"find\"";
+    }
```
Comment (chinhan99) [inline src/main/java/errors/ErrorMessages.java]: Maybe can use enum in the future instead. It should work similarly with lesser lines.

File: src/main/java/util/Storage.java [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+            try {
+                appendToFile(currentLine);
+            } catch (IOException e) {
+                System.out.println(ErrorMessages.provideUnableToWriteToFileText());
+            }
+        }
+    }
+
+    private static String parseString(Todo todo) {
+        String done = "0";
```
Comment (chinhan99) [inline src/main/java/util/Storage.java]: Might be able to change some of these magic literals to constants

File: src/main/java/errors/ErrorMessages.java [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+            return "PIKAPII you dont have that many tasks! You only have 1 task hehe :3";
+        }
+        return "PIKAPII you dont have that many tasks! Please write a number between 1 and " + listLength;
+    }
+
+    public String provideInvalidNumberText() {
+        return "PIKAPII we dont have negative or tasks indexed as 0 >:<";
+    }
+
+    public String provideNoNumberText() {
```
Comment (chinhan99) [inline src/main/java/errors/ErrorMessages.java]: The naming of this method could be more descriptive , to better describe that it is an error message method

**Review by chinhan99 — COMMENTED** [review]
1) Do consider changing the name of some of your methods to better describe it's purpose. 
2) There are still some magic literals within your code, maybe you can consider swapping tham with constants in there. 
3) More comments can be inserted, especially for method header comments. 

Overall good work. Noticed OOP put to practice.

**Pull request closed without merging**